cmake/FindUnbound.cmake

```diff
@@ -0,0 +1,56 @@
+# FindUnbound.cmake - Find libunbound package.
+#
+# Copyright 2023 Intel Corporation
+# SPDX-License-Identifier: Apache 2.0
+#
+
+#-----------------------------------------------------------------------
+# Use pkg-config to search for the module.
+#-----------------------------------------------------------------------
+find_package(PkgConfig)
+
+pkg_check_modules(UNBOUND QUIET libunbound)
+
+#-----------------------------------------------------------------------
+# Find libraries and include directories.
+#-----------------------------------------------------------------------
+find_path(UNBOUND_INCLUDE_DIR
+    NAMES "unbound.h"
+    PATHS ${UNBOUND_INCLUDE_DIRS}
+)
+
+find_library(UNBOUND_LIBRARY
+    NAMES unbound
+    PATHS ${UNBOUND_LIBRARY_DIRS}
+)
+
+#-----------------------------------------------------------------------
+# Get version number
+#-----------------------------------------------------------------------
+if(UNBOUND_VERSION)
+    set(UNBOUND_VERSION ${UNBOUND_VERSION} CACHE STRING "libunbound version")
+endif()
+
+#-----------------------------------------------------------------------
+# Handle REQUIRED and QUIETLY arguments
+#-----------------------------------------------------------------------
+find_package_handle_standard_args(UNBOUND
+    REQUIRED_VARS
+        UNBOUND_LIBRARY
+        UNBOUND_INCLUDE_DIR
+    VERSION_VAR
+        UNBOUND_VERSION
+)
+
+mark_as_advanced(UNBOUND_INCLUDE_DIR UNBOUND_LIBRARY UNBOUND_VERSION)
+
+#-----------------------------------------------------------------------
+# Define library target
+#-----------------------------------------------------------------------
+add_library(unbound UNKNOWN IMPORTED)
```

Comment:
should we add in our document that libunbound should be manually installed?
or is it part of legacy OVS doc?

Reply:
It's really a function of what features of OVS the customer wants to use or support. We're just adapting to it by providing access to the library when it exists.

We've been running without `libunbound` for over a year. It only became an issue when Sabeel built P4 Control Plane on a system that actually had it installed. The OVS build detected the installation and enabled conditional code to use it, which broke our build because we didn't have support for the library in our build system.

For the time being, I think we can defer to the OVS documentation, but you raise an excellent point. (hat tip.)  To serve the needs of our customers, we should have a solid understanding of the requirements (both mandatory and optional) of all the packages we supply or support; we should provide for them in our build systems; and we should consider them in our setup procedures, validation procedures, and documentation.